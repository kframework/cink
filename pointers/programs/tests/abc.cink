/*
  For symbolic execution of the program:
    -- kompile cink --backend symbolic
    -- krun programs/abc.cink -cIN="ListItem(#symInt(a))" -cPC="true"
    or
    -- krun programs/abc.cink -cIN="ListItem(#symInt(a))" -cPC="true" --search 
*/
int main() {
  int *a = new int;
  int *b = new int;
  int *c = new int;
  cin >> *a;
  *b = (*a)++;
  if (*a == (*b) + 1) cout << "a=b" << endl;
  *c = ++(*b) + 1;
  if (*c == (*b) + 1) cout << "c = b+1" << endl;
}

