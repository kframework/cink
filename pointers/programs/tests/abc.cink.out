a=b
c = b+1
<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            noVal
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            main |-> #symScalLoc(1)
        </env>
    </control>
    <genv>
        main |-> #symScalLoc(1)
    </genv>
    <store>
        #symScalLoc(1) |-> lambda (  @ a of pointer to int ; (a = (new int
          )) ; b of pointer to int ; (b = (new int)) ; c of pointer to int ; (
          c = (new int)) ; (cin >> (* a)) ; ((* b) = ((* a) ++)) ; if ( ((* a)
           == ((* b) + 1)) ) ((cout << "a=b") << (endl)) ; else { } ((* c) = (
          (++ (* b)) + 1)) ; if ( ((* c) == ((* b) + 1)) ) ((cout << "c = b+1"
          ) << (endl)) ; else { } )
        #symScalLoc(2) |-> #symScalLoc(3)
        #symScalLoc(3) |-> 8
        #symScalLoc(4) |-> #symScalLoc(5)
        #symScalLoc(5) |-> 8
        #symScalLoc(6) |-> #symScalLoc(7)
        #symScalLoc(7) |-> 9
    </store>
</T>
